src/routes/auth/auth-controller.ts

```diff
@@ -0,0 +1,91 @@
+import { NextFunction, Response, Request } from "express";
+import { authService } from "../../services/auth-service";
+import { errors } from "../../utils/errors";
+import { User } from "../../models/user-model";
+import { BodyRequest } from "../../@types/request-types";
+import { jwtService } from "../../services/jwt-service";
+import bcrypt from "bcrypt";
+
+interface AuthResponse {
+  user: User;
+  authToken: string;
+}
+
+export const authController = {
+  login: async (
+    req: BodyRequest<User>,
+    res: Response<AuthResponse>,
+    next: NextFunction
+  ) => {
+    try {
+      const user = req.body;
+      const storedUser = await authService.findByEmail(user.email);
+      if (!storedUser) {
+        return next(errors.unsuccesfullLogin("incorrect email"));
+      }
+      const isPasswordMatching = await bcrypt.compare(
+        user.password,
+        storedUser.password
+      );
+      if (!isPasswordMatching) {
+        return next(errors.unsuccesfullLogin("incorrect password"));
```

Comment:
In this case (login) for security reason is a good practices to not specify which field is wrong please use a more generic error for both email and password verification like "Wrong credentials" or "Wrong email or password" otherwise easily can be know if an email exist in the sistem or even worst if a password matching